staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -490,3 +511,58 @@ func startDBSizeMonitorPerEndpoint(client *clientv3.Client, interval time.Durati
 		cancel()
 	}, nil
 }
+
+// checkVersion checks for deprecated versions of etcd and returns an error if find any
+// currently the versions < 3.4.25, between 3.5.0 and 3.5.7
+// based on this: https://github.com/etcd-io/etcd/issues/15220
+func checkVersion(c *clientv3.Client, dialTimeOut time.Duration) (err error) {
+	eps := c.Endpoints()
+	baseCtx := context.TODO()
+
+	ctx, cancel := context.WithCancel(baseCtx)
```

Comment:
This is not needed, errgroup will handle closing context when it exits.

Reply:
Ok i have removed the cancels, and add a comment to point out this, thanks